app/src/test/java/com/example/memeitupapp/memedetailtest/memedialogmodeltest/MemeDetailModelTest.kt

```diff
@@ -18,32 +20,38 @@ class MemeDetailModelTest {
     @get:Rule
     val instantTaskExecutorRule = InstantTaskExecutorRule()
 
-    private val memeService: MemeService = mock()
+    private val mockedGetMemeByIdUseCase: GetMemeByIdUseCase = mock()
+    private val mockedGetMemeByIdFromDataBaseUseCase: GetMemeByIdFromDataBaseUseCase = mock()
+    private val mockedUpdateMemeDetailsDataBaseUseCase: UpdateMemeDetailsDataBaseUseCase = mock()
     private lateinit var memeDetailModel: MemesDetailsContract.Model
 
     @Before
     fun setup() {
-        memeDetailModel = MemeDetailModel(memeService)
+        memeDetailModel = MemeDetailModel(mockedGetMemeByIdUseCase, mockedUpdateMemeDetailsDataBaseUseCase, mockedGetMemeByIdFromDataBaseUseCase)
     }
 
     @Test
-    fun `on get meme details successfully`() {
+    fun `on GetMemeByIdUseCase gets data from API, updates data base`() {
         val mockedMemeResult: Result.Success<MemeDetail> = mock()
-        whenever(memeService.getMemeByIdFromApi(MEME_ID)).thenReturn(mockedMemeResult)
-
-        runBlocking { memeDetailModel.getMeme(MEME_ID) }
+        whenever(mockedGetMemeByIdUseCase.invoke(MEME_ID)).thenReturn(mockedMemeResult)
 
         assertEquals(mockedMemeResult, memeDetailModel.getMeme(MEME_ID))
+
+        verify(mockedGetMemeByIdUseCase).invoke(MEME_ID)
+        verify(mockedUpdateMemeDetailsDataBaseUseCase).invoke(mockedMemeResult.data)
     }
 
     @Test
-    fun `on get meme details with error connection`() {
-        val mockedMemeResult: Result.Failure = mock()
-        whenever(memeService.getMemeByIdFromApi(MEME_ID)).thenReturn(mockedMemeResult)
+    fun `on GetMemeByIdUseCase fails, invoke GetMemeByIdFromDataBase`() {
+        val mockedMemeDetailResultFailure: Result.Failure = mock()
```

Comment:
you can omit this variable because you don't do any assert with it

Reply:
You are right, deleted!